### PR TITLE
WIP: Allow custom layout for redirect page

### PR DESCRIPTION
**Pull request by pathawks**
- [x] Allow user to override layout template
- [ ] Add test for new feature
- [x] Add docs for new feature

### CODE REVIEW

File: lib/jekyll-redirect-from/redirect_page.rb [hunk 1]
```diff
@@ -19,16 +19,33 @@ def initialize(site, base, dir, name)
...
+    def info
+      {
+        :filters   => [Jekyll::Filters]
+      }
+    end
+
+    def template
+      @template_contents unless @template_contents.nil?
+      if @site.layouts.key?(@site.config['redirect_template'])
+        @template_contents = Liquid::Template.parse(@site.layouts[@site.config['redirect_template']].content)
```
Comment (parkr) [inline lib/jekyll-redirect-from/redirect_page.rb]: I'd really rather not add another config var. :christmas_tree:

File: lib/template.html [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<!DOCTYPE html>
+<meta charset="utf-8">
+<title>Redirecting…</title>
+<link rel="canonical" href="{{ page.url }}">
+<meta http-equiv="refresh" content="0; url={{ page.url }}">
+<h1>Redirecting…</h1>
+<a href="{{ page.url }}">Click here if you are not redirected.</a>
+<script>location="{{ page.url }}"</script>
```
Comment (parkr) [inline lib/template.html]: Let's mimic Jekyll-sitemap and put this in lib/jekyll-redirect-from.

File: lib/jekyll-redirect-from/redirect_page.rb (hunk 1 above)
Comment (pathawks) [inline lib/jekyll-redirect-from/redirect_page.rb]: Look for a *special* layout called `redirect` instead?

**Comment by pathawks** [comment]
Currently, this will use a layout named `redirect` if it exists, falling back to the old template in `jekyll-redirect-from.html` if it does not.

I am unsure of how to write a test for this, without setting up a second fixture site. Either the layout exists or it doesn't. :confused:

**Comment by parkr** [comment]
@pathawks It's fine to setup a fixture site if you'd like! Integration tests like that are a bit slower but it more directly maps the user experience :smile:

**Comment by parkr** [comment]
@pathawks Any update here?

**Comment by pathawks** [comment]
Sorry, no.

I'll take a look at this again tomorrow.

**Comment by benbalter** [comment]
@pathawks what's your motivation/use case here? When would you see a user doing this? To brand the HTML that flashes on the screen for a second?

**Comment by pathawks** [comment]
> what's your motivation/use case here?

If I'm being totally honest, I think it is a little bit silly. I think the primary goal of the plugin should be to get the view from the old page to the new page as quickly as possible.

That being said, there have been a number of requests to add things to the redirect template that would *not* be beneficial to the vast majority of users of the plugin, but some people really really want these things anyway.
See #55 #63 #91

This PR will attempt to address @parkr's comment [here](https://github.com/jekyll/jekyll-redirect-from/issues/55#issuecomment-170392837)

> I think allowing the user to specify a `_layouts/jekyll-redirect-from.html` would be ideal. I'd strongly consider a PR which allowed the author to specify a layout and use the default boring one if none is suggested.

The downside with allowing users to specify their own template is that if down the road we discover something that we should really implement on all redirect pages, we will not be able to roll it out to all users: some users will need to manually update their redirect templates.

So, obviously I'm conflicted here. This is not something that I would use myself.

**Comment by parkr** [comment]
> what's your motivation/use case here?

@benbalter: The motivation is similar to the idea behind allowing folks to override feed.xml or sitemap.xml, but in this case it's just putting a nice wrapper around our file. Instead of generating the entire HTML, we'd specify a default layout that could be overridden in the user's repo so they can style this page how they want. Folks are upset that visitors see a gross white page and want to fix that. This is one way of solving that issue.

Another way would be to reach down to the server and provide a list of URL mapping that could give a _true_ 301 redirect to the right URL.

**Comment by benbalter** [comment]
> we'd specify a default layout that could be overridden in the user's repo so they can style this page how they want. Folks are upset that visitors see a gross white page and want to fix that. This is one way of solving that issue

Makes sense. Thanks for the context.

**Comment by parkr** [comment]
@pathawks Would love to merge and release this 💓

**Comment by pathawks** [comment]
After reading [**Removing a feature is a feature**](http://ben.balter.com/2016/07/21/removing-a-feature-is-a-feature/) and thinking again about [**Optimizing for power users and edge cases is the easy part**](http://ben.balter.com/2016/03/08/optimizing-for-power-users-and-edge-cases/), I've decided to close this PR.

This feature is not something I would use, and not something I am excited about supporting. I think the value of this plugin is in providing a simple way for users to build redirects. There are already other ways for users to do that, the focus here needs to be on simplicity.

If a user was so concerned with making redirect pages look a certain way, they could bypass this plugin entirely and just have a simple redirect layout specific to their needs.

If somebody else wants to take this ball and run with it, that is fine by me. The only thing missing from this PR is tests. Personally, I would just feel icky if pushing a feature that I would never use myself and had little interest in providing support for.

**Pull request closed without merging**